packages/client/lib/rpc/modules/engine.ts

```diff
@@ -513,10 +513,16 @@ export class Engine {
       await this.execution.setHead(headBlock)
       this.txPool.removeNewBlockTxs(blocks)
 
-      const timeDiff = new Date().getTime() / 1000 - headBlock.header.timestamp.toNumber()
+      // TODO: removed along rebase, 2022-04-05, @holgerd77
+      // Analyze if code should be there for some reason
+      /*for (const block of blocks) {
+        this.validBlocks.delete(block.hash().toString('hex'))
+      }*/
```

Comment:
I wasn't 100% sure if this code part should be there or not, guess @ryanio or @g11tech will likely know.

Reply:
That can be removed, I removed validBlocks (we store them in the blockchain directly now)

Reply:
I think this is needed, because @ryanio has been maintaining the validBlocks set of the blocks which have valid execution, but not yet part of the chain, since here fcU has arrived for those blocks, they are being moved from validBlock to the main chain via putBlocks.